internal/ui/model/chat.go

```diff
@@ -98,24 +169,32 @@ func (m *Chat) Blur() {
 	m.list.Blur()
 }
 
-// ScrollToTop scrolls the chat view to the top.
-func (m *Chat) ScrollToTop() {
+// ScrollToTop scrolls the chat view to the top and returns a command to restart
+// any paused animations that are now visible.
+func (m *Chat) ScrollToTop() tea.Cmd {
 	m.list.ScrollToTop()
+	return m.RestartPausedVisibleAnimations()
 }
 
-// ScrollToBottom scrolls the chat view to the bottom.
-func (m *Chat) ScrollToBottom() {
+// ScrollToBottom scrolls the chat view to the bottom and returns a command to
+// restart any paused animations that are now visible.
+func (m *Chat) ScrollToBottom() tea.Cmd {
 	m.list.ScrollToBottom()
+	return m.RestartPausedVisibleAnimations()
 }
 
-// ScrollBy scrolls the chat view by the given number of line deltas.
-func (m *Chat) ScrollBy(lines int) {
+// ScrollBy scrolls the chat view by the given number of line deltas and returns
+// a command to restart any paused animations that are now visible.
+func (m *Chat) ScrollBy(lines int) tea.Cmd {
 	m.list.ScrollBy(lines)
+	return m.RestartPausedVisibleAnimations()
 }
 
-// ScrollToSelected scrolls the chat view to the selected item.
-func (m *Chat) ScrollToSelected() {
+// ScrollToSelected scrolls the chat view to the selected item and returns a
+// command to restart any paused animations that are now visible.
+func (m *Chat) ScrollToSelected() tea.Cmd {
 	m.list.ScrollToSelected()
+	return m.RestartPausedVisibleAnimations()
```

Comment:
Let's keep the logic here separate and instead use `RestartPausedVisibleAnimations` whenever needed in Update

Reply:
Why tho? this is the normal scroll, BUT for the chat component the behavior is that on every scroll we make sure we reset visible animations 🤔